### DISCUSSION TITLE
My FlySky FS-HZCZ03-ADJ Digital Gimbal is not working

### DISCUSSION THREAD

**Discussion by Snekovo** [OpenI6X/opentx#416] [Q&A] [answered]
Does the FS-HZCZ03-ADJ V2 gimbal work for anyone?

I have connected according to the modification instructions
RXD4 <----> FS-HZCZ03-ADJ Gimbal TX
3V3 <----> FS-HZCZ03-ADJ Gimbal 3V3
GND <----> FS-HZCZ03-ADJ Gimbal GND

The gimbal should automatically detect it and if not, it will switch to a classic analog gimbal.

I have firmware version OpenI6X v1.11.1 and auto-detection will not occur

the script /tools/build-flysky.py does not contain the DFLYSKY_GIMBAL option listed in radio/src/targets/flysky/CMakeLists.txt
I don't know cmake and I'm not a programmer

the compiler command does not take over the parameter DFLYSKY_GIMBAL=YES

_docker run --rm -it -e "BOARD_NAME=i6x" -e "CMAKE_FLAGS=PCB=I6X DFLYSKY_GIMBAL=YES HELI=NO GVARS=YES LUA_COMPILER=NO MULTIMODULE=NO PCBI6X_ELRSV3=NO DEBUG=NO TRANSLATIONS=CZ" -v ${PWD}:/opentx ajjjjjjjj/opentx-docker-i6x_

if I add to build-flysky.py section common_options "DFPLAYER": "YES"
it also doesn't detect gimbals

what am I doing wrong?

![Obrázek WhatsApp, 2024-08-26 v 20 04 46_168d619b](https://github.com/user-attachments/assets/64e4d43d-e26a-4f58-ac22-c85a753dc5ba)

**Comment by ajjjjjjjj** [comment, marked as answer]
Try removing "D": `FLYSKY_GIMBAL=YES`, also `PCBI6X_ELRSV3` is wrong and should be `PCBI6X_ELRS`. 
I have updated documentation.

**Reply by Snekovo** [reply]
Thank you, now the gimbals works . 
If you are interested, I can prepare a "photo documentation" of the installation.

**Reply by ajjjjjjjj** [reply]
Sure, will be great.

**Comment by Snekovo** [comment]
Note, HAL gimbal autodetection does not work on OpenI6X when the firmware is recompiled with HAL support , returned original analog gimbals do not work.

**Reply by ajjjjjjjj** [reply]
You did not provided any information on build time opions or branch/tag used and how is it different from previous success in getting those to work.

**Reply by Snekovo** [reply]
Version OpenI6X v1.11.1 and FLYSKY_GIMBAL=YES 

not valid information from  https://github.com/EdgeTX/edgetx/wiki/Flysky-Hall-Sticks-Mod

_The code includes an automatic detection of FlySky digital hall gimbals on power-up of TX16S/T16/T18 radios and no special build-time option is required. If after initial 70 ms the FlySky gimbals are not detected, the code reverts to sampling analog sticks._

I have two radios with different gimbals, verified with the same firmware.

I wouldn't address this feature, just listing how it works for information. :)

**Reply by ajjjjjjjj** [reply]
ahh, ok, that's true :) It's not built in due to FLASH size constrains.
